### ISSUE TITLE
chore(website): Replace subscribe serverless function with client side subscribe

### ISSUE THREAD

**Issue by erezrokah** [cloudquery/cloudquery#2682]
## Problem description

On our Website we have a form to subscribe to our newsletter.
It uses a Serverless Vercel function to handle the subscribe operation:
https://github.com/cloudquery/cloudquery/blob/10be7695e6164ed099c316b531ff6166724cc452/website/pages/api/subscribe.ts#L13

Since the function uses secrets we set those as environment variables in the Vercel project configuration.
As a result any PR from [a fork needs to be authorized by a team member.](https://vercel.com/docs/concepts/git#deploying-forks-of-public-git-repositories)

## TLDR

Having a subscribe function adds friction to every PR from a forked repository

## Solution

We don't need a function to handle the subscribe operation, as we can do it in the client side, see https://mailchimp.com/help/host-your-own-signup-forms/.

Once we do that we can remove the environment variables from the Vercel project and forked PRs should not require authorization
